citar.el

```diff
@@ -703,7 +699,7 @@ key associated with each one."
          (raw-candidates
           (parsebib-parse bib-files :fields (citar--fields-to-parse)))
          (hasfilep (citar-has-file))
-         (hasnote (citar-keys-with-notes))
+         (hasnote (citar-has-note))
```

Comment:
Should probably call this `hasnotep` just like `hasfilep`.

Also, on line 710 below, should be `(funcall hasnotep citekey entry)` instead of `(member ...)`.

Reply:
Is there a reason not to use hypens; `has-note-p`?

Reply:
Yeah, local variables [should not use hyphens in their names](https://with-emacs.com/posts/tutorials/almost-all-you-need-to-know-about-variables/#:~:text=One%20trick%20to%20prevent%20such%20conflicts%20is%20to%20avoid%20local%20variables%20which%20have%20dashes%20in%20their%20name.).

That's because in modern elisp `let` does lexical binding instead of dynamic binding, _unless_ the variable in question has a global definition. To make sure this can't happen, the convention is that all global variables have at least one hyphen (which is usually the case) whereas all local variables introduced in `let` bindings have no hyphens.

Reply:
Yeah, just keep it in mind and slowly change over if/when you touch some piece of code in the future. Since this only applies to local variables anyway, it can't really cause breaking changes outside the scope of a single function.